calculator.c

```diff
@@ -0,0 +1,429 @@
+		return -1; //doesnt matter, in this case ERROR_CODE flage is raised
+	}
+	return (*fun_ptr_arr[rightOerationNumber])(first, second);
+}
+
+
+/**
+ *  This fuction takes a string as input,
+ *  and return is as a uint64_t int32_t.
+ *  it also checks if it is really a number represented in a string,
+ *  if not it raises flag ERROR_CODE and returns -1 that will not be taken into account
+ *  @param char* stringNbr, the string to be parsed into a uint64_t int32_t.
+ *  @return the parsed string as a uint64_t int32_t or -1 and raise flage ERROR_CODE.
+ */
+uint64_t stringConvertToLong(char* stringNbr) {
+	uint64_t number = 0;
+
+	for (char* digit = stringNbr; *digit; ++digit)
+	{
+		if( *digit >= '0' && *digit <= '9' ){
+		number = number * 10 + *digit - '0';
+		}
+		
+		else{
+			//stringNbr is not really a number
+			ERROR_CODE = invalidInput;
+			return -1;//doesnt matter
+		}
+	}
+	return number;
+}
+/**
+ *  This fuction takes a two uint64_t ints as input,
+ *  and checks if they can be contained in an int32_t if so, it returns 1.
+ *  if one of the inputs int32_t-overflows, it returns 0
+ *  @param firstLongNbr and secondLongNbr, the number that we check the dont int32_t-overflow.
+ *  @return 1 if the numbers are not int32_t-overflowing, 0 else
+ */
+int32_t verifyNumbersAreInt(uint64_t firstLongNbr, uint64_t secondLongNbr) {
+	return !(firstLongNbr > INT_MAX || secondLongNbr > INT_MAX);
+}
+
+
+/**
+ *  This fuction takes as input : the two input as string : firstStr and secondStr
+ *  plus two int32_t buffer : firstNbrInt and secondNbrInt that are passed by reference that
+ *  will contain the int32_t parsed value of the the input strng respectively, in case the two string
+ *  are really string representation of numbers, not longer than MAXSIZEINPUT, not negative,
+ *  whole numbers and not int32_t-Overflowing
+ *  If the two input strings respect all the rules that we just described, it put their value as
+ *  int32_t the buffers firstNbrInt and secondNbrInt and return 1 else 0.
+ *  if one of the inputs int32_t-overflows, it returns 0
+ *  @param "firstStr" and "secondStr" the string to check if their parsed values can be contained 
+ *  in an int32_t and if they are valid, we put the int32_t representation of firstStr in the input int32_t buffer "firstNbrInt",
+ *  the same for secondStr that we put in "secondNbrInt"
+ *  @return 1 if firstStr and secondStr can be parsed into an int32_t, 0 else
+ */
+int32_t inputIsValid(char* firstStr, char* secondStr, int32_t* firstNbrInt, int32_t* secondNbrInt) {
+	//pointer to fucntion declaration
+	int32_t (*ptrVerifyNumbersAreInt)(uint64_t, uint64_t) = &verifyNumbersAreInt;
+	uint64_t (*ptrStringConvertToLong)(char*) = &stringConvertToLong;
+
+	
+	if (strnlen(firstStr, MAXSIZEINPUT + 1) > MAXSIZEINPUT
+		|| strnlen(secondStr, MAXSIZEINPUT + 1) > MAXSIZEINPUT) {
+		ERROR_CODE = invalidInput;
+		return 0;
+	}
+	//we dont support negative numbers
+	if (*firstStr == '-' || *secondStr == '-') {

```

Comment:
If you want to check the first char in a string (or generally the Xth char), use `str[x]` instead of `*str` (or `*str + x`)